crates/pathfinder/src/state/sync/pending.rs

```diff
@@ -16,20 +19,65 @@ use crate::state::sync::SyncEvent;
 /// - the state update parent root does not match head.
 ///
 /// A full block or full state update can be returned from this function if it is encountered during polling.
-pub async fn poll_pending(
+pub async fn poll_pending<S: GatewayApi + Clone + Send + 'static>(
     tx_event: tokio::sync::mpsc::Sender<SyncEvent>,
-    sequencer: impl GatewayApi + Clone + Send + 'static,
+    sequencer: &S,
     head: (
         pathfinder_common::BlockHash,
         pathfinder_common::StateCommitment,
+        pathfinder_common::StarknetVersion,
     ),
     poll_interval: std::time::Duration,
     storage: Storage,
 ) -> anyhow::Result<(Option<Block>, Option<StateUpdate>)> {
-    use anyhow::Context;
-    use pathfinder_common::BlockId;
-    use std::sync::Arc;
+    const V_0_12_1: semver::Version = semver::Version::new(0, 12, 1);
+    let (block_hash, state_commitment, starknet_version) = head;
+    match starknet_version.parse_as_semver() {
+        Ok(Some(version)) => match version {
+            v if v > V_0_12_1 => {
```

Comment:
Technically incorrect if they introduce v0.12.1.1 to mainnet or elsewhere in the meantime. Ofc correct in a strict semver case.

Reply:
That's not really a problem -- `parse_as_semver()` truncates the version string to the first 3 segments, so 0.12.1.1 will still parse as 0.12.1, which will be less than `V_0_12_2`.

Reply:
I blame my vacation brain.